### ISSUE TITLE
Suspicious Kernel Messages

### ISSUE THREAD

**Issue by marwinski** [gardenlinux/gardenlinux#945]
**What happened**:

I do see the following kernel messages / warnings / errors that appear to indicate underlying issues:

1) Many occurrences on all platforms of the messages below. Likely to be caused by Kubernetes but could also be triggered by a different reason.

```
overlayfs: lowerdir is in-use as upperdir/workdir of another mount, accessing files from both mounts will result in undefined behavior.
overlayfs: upperdir is in-use as upperdir/workdir of another mount, accessing files from both mounts will result in undefined behavior.
overlayfs: workdir is in-use as upperdir/workdir of another mount, accessing files from both mounts will result in undefined behavior.
```

2) I saw this (only) once on VMware with Garden Linux 576.3:

```
[1550380.499157] TCP: eth0: Driver has suspect GRO implementation, TCP performance may be compromised.
```

3) systemd-journald core dump. Saw this with 576.3 on VMware and AWS:

```
[2397546.394387] systemd[1]: promtail-fetch-token.service: Deactivated successfully.
[2397546.665786] systemd-coredump[2289464]: Process 471 (systemd-journal) of user 0 dumped core.
[2397546.667741] systemd-coredump[2289464]: Coredump diverted to /var/lib/systemd/coredump/core.systemd-journal.0.1b47b5837e03491aa201852a0e197c8e.471.1652999821000000.zst
[2397546.671271] systemd-coredump[2289464]: Found module linux-vdso.so.1 with build-id: 6cd7bd966ba94e5ce86493a13896cb4ce41777a0
[2397546.674054] systemd-coredump[2289464]: Found module libcap-ng.so.0 with build-id: da5dd811e7ae08d73c47c4d7683f4dd8cd475d47
[2397546.676504] systemd-coredump[2289464]: Found module libaudit.so.1 with build-id: f5e9abe3d5cfa453650f85fd7b1619a2327e4199
[2397546.679010] systemd-coredump[2289464]: Found module libcrypto.so.1.1 with build-id: c0211af14123df3689c3881301b7ef13c7cd090a
[2397546.681691] systemd-coredump[2289464]: Found module libgpg-error.so.0 with build-id: 98f570f9581025c47376987213093e37a9fc36c3
[2397546.684375] systemd-coredump[2289464]: Found module libpcre2-8.so.0 with build-id: cb2f45432240ff04fd5089cd02fef822a46a35b7
[2397546.688202] systemd-coredump[2289464]: Found module ld-linux-x86-64.so.2 with build-id: 6211a5e83642f3c0cb0b1670ee201d1d9d72e05e
[2397546.690475] systemd-coredump[2289464]: Found module libpthread.so.0 with build-id: bfc63b7599bf70fd14c5520befdcdabb16840416
[2397546.692611] systemd-coredump[2289464]: Found module liblzma.so.5 with build-id: 5d67991a152e0b62f982a0e4110cc2262850c788
[2397546.695834] systemd-coredump[2289464]: Found module libzstd.so.1 with build-id: 57509eed80c587b9ef77d3d1234e969b98448664
[2397546.698306] systemd-coredump[2289464]: Found module libseccomp.so.2 with build-id: 1fd882f98ed339a6d85c7a161f15ca145464b523
[2397546.700882] systemd-coredump[2289464]: Found module librt.so.1 with build-id: e2d858f8549e26d34d4bcb33e29f24d084523c68
[2397546.703340] systemd-coredump[2289464]: Found module libpam.so.0 with build-id: 852dcb0ddb31904b52c3532beaba1013d4341130
[2397546.706114] systemd-coredump[2289464]: Found module libmount.so.1 with build-id: 60ed2677f96dde3b2a989046ec80cb8b5cbfc59a
[2397546.708660] systemd-coredump[2289464]: Found module liblz4.so.1 with build-id: 964039e18af4b59e5a11f4ad26e9aa5e6a2d5db7
[2397546.711212] systemd-coredump[2289464]: Found module libkmod.so.2 with build-id: 8efc2d78710d7791d15098af30e9ef2ee2029f0b
[2397546.713805] systemd-coredump[2289464]: Found module libip4tc.so.2 with build-id: 386795a5ea444e162fb9733ebf31201fbe23a691
[2397546.716180] systemd-coredump[2289464]: Found module libgcrypt.so.20 with build-id: a4be95e16dd37a01c63de407f0f42ec8a200ee6a
[2397546.719594] systemd-coredump[2289464]: Found module libdl.so.2 with build-id: e3eb1a873134b05c621c37b47d8a7d94ca31ea74
[2397546.723690] systemd-coredump[2289464]: Found module libcrypt.so.1 with build-id: f3777a1beeddd86c72488870aef65d0b425ad4af
[2397546.731208] systemd-coredump[2289464]: Found module libcap.so.2 with build-id: a6034f7fc277ee9d9714c2b288b419498225156f
[2397546.734529] systemd-coredump[2289464]: Found module libblkid.so.1 with build-id: d1947ab0e5ecfac0693932328e3a3b1c8a56d22c
[2397546.736602] systemd-coredump[2289464]: Found module libacl.so.1 with build-id: 10f984c014a2f7b28613cd44a98cf1d2e4a5eb24
[2397546.738656] systemd-coredump[2289464]: Found module libc.so.6 with build-id: 3a69683d31c430fad5cb0fad190a28b9570d5577
[2397546.741453] systemd-coredump[2289464]: Found module libselinux.so.1 with build-id: cc3fa4080d349d749e3045798819b0b5299618b0
[2397546.743782] systemd-coredump[2289464]: Found module libsystemd-shared-249.so with build-id: f50a6ceec4d600f67d20b89179b6ba1533a37343
[2397546.746497] systemd-coredump[2289464]: Found module systemd-journald with build-id: 49f7b17fb16f685023445deb28698ec018e8ec26
[2397546.748665] systemd-coredump[2289464]: Stack trace of thread 471:
[2397546.750033] systemd-coredump[2289464]: #0  0x00007efcef6493fe n/a (libsystemd-shared-249.so + 0x1cb3fe)
[2397546.751863] systemd-coredump[2289464]: #1  0x00007efcef64a7f0 journal_file_append_entry (libsystemd-shared-249.so + 0x1cc7f0)
[2397546.766900] systemd-coredump[2289464]: #2  0x000055d280e7d157 n/a (systemd-journald + 0x10157)
[2397546.768791] systemd-coredump[2289464]: #3  0x000055d280e80b2e n/a (systemd-journald + 0x13b2e)
[2397546.772795] systemd-coredump[2289464]: #4  0x000055d280e80e1e n/a (systemd-journald + 0x13e1e)
[2397546.775192] systemd-coredump[2289464]: #5  0x000055d280e8155a n/a (systemd-journald + 0x1455a)
[2397546.779325] systemd-coredump[2289464]: #6  0x000055d280e81ad2 n/a (systemd-journald + 0x14ad2)
[2397546.781142] systemd-coredump[2289464]: #7  0x00007efcef66c710 n/a (libsystemd-shared-249.so + 0x1ee710)
[2397546.783655] systemd-coredump[2289464]: #8  0x00007efcef66c9fd sd_event_dispatch (libsystemd-shared-249.so + 0x1ee9fd)
[2397546.787164] systemd-coredump[2289464]: #9  0x00007efcef66dfd0 sd_event_run (libsystemd-shared-249.so + 0x1effd0)
[2397546.790032] systemd-coredump[2289464]: #10 0x000055d280e7540d n/a (systemd-journald + 0x840d)
[2397546.791718] systemd-coredump[2289464]: #11 0x00007efcef2b0e4a __libc_start_main (libc.so.6 + 0x27e4a)
[2397546.793662] systemd-coredump[2289464]: #12 0x000055d280e75aea n/a (systemd-journald + 0x8aea)
[2397546.804141] systemd[1]: systemd-journald.service: Main process exited, code=dumped, status=6/ABRT
[2397546.805933] systemd[1]: systemd-journald.service: Failed with result 'watchdog'.
[2397546.811657] systemd[1]: systemd-journald.service: Scheduled restart job, restart counter is at 2.
[2397546.818763] systemd[1]: Stopping Flush Journal to Persistent Storage...
[2397546.830700] systemd[1]: systemd-journal-flush.service: Deactivated successfully.
[2397546.834835] systemd[1]: Stopped Flush Journal to Persistent Storage.
[2397546.836809] systemd[1]: Stopped Journal Service.
[2397546.843018] systemd[1]: Starting Journal Service...
[2397546.901445] systemd-journald[2289637]: File /var/log/journal/afa6c8f15048456e88579d360cc4dbb0/system.journal corrupted or uncleanly shut down, renaming and replacing.
[2397546.945367] systemd[1]: Started Journal Service.
[2397548.030002] sd 0:0:0:0: [sda] tag#113 task abort on host 0, 000000000ac03e51
```

4) Some ugly ext4 errror (AWS, 576.3):

```
*** lots of the following, fluentd bug
[174861.000540] EXT4-fs warning (device nvme7n1): ext4_dx_add_entry:2355: Large directory feature is not enabled on this filesystem
[175372.360079] ------------[ cut here ]------------
[175372.363691] bdi-(unknown) not registered
[175372.367010] WARNING: CPU: 3 PID: 1118335 at fs/fs-writeback.c:2307 __mark_inode_dirty+0x247/0x2f0
[175372.373876] Modules linked in: wireguard libchacha20poly1305 chacha_x86_64 poly1305_x86_64 ip6_udp_tunnel udp_tunnel libblake2s blake2s_x86_64 curve25519_x86_64 libcurve25519_generic libchacha libblake2s_generic ipt_rpfilter xt_multiport xt_set iptable_raw ip_set_hasd
[175372.373928]  cryptd glue_helper serio_raw autofs4
[175372.433149] CPU: 3 PID: 1118335 Comm: kubelet Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[175372.441081] Hardware name: Amazon EC2 m5.2xlarge/, BIOS 1.0 10/16/2017
[175372.445437] RIP: 0010:__mark_inode_dirty+0x247/0x2f0
[175372.449138] Code: ff ff 48 8b 38 48 89 c5 f6 47 44 01 74 1e 48 8b 40 08 a8 01 75 16 e8 18 ed f3 ff 48 c7 c7 a3 bb aa be 48 89 c6 e8 c3 e6 54 00 <0f> 0b 48 8b 05 40 4f f2 00 49 89 85 c8 00 00 00 45 85 ff 74 0e 48
[175372.461543] RSP: 0018:ffff9cbb40d47d48 EFLAGS: 00010286
[175372.465363] RAX: 0000000000000000 RBX: 0000000000000003 RCX: 0000000000000000
[175372.471851] RDX: ffff9107e26ec0e0 RSI: ffff9107e26dc380 RDI: ffff9107e26dc380
[175372.478074] RBP: ffff9101cf737860 R08: 0000000000000000 R09: ffff9cbb40d47b68
[175372.484232] R10: ffff9cbb40d47b60 R11: ffffffffbecc8908 R12: ffff9103caaa0668
[175372.490347] R13: ffff9103caaa05e0 R14: 0000000000000000 R15: 0000000000000000
[175372.496612] FS:  00007f41f67fc640(0000) GS:ffff9107e26c0000(0000) knlGS:0000000000000000
[175372.503200] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[175372.507247] CR2: 00007ff42e389300 CR3: 00000003ac3d0006 CR4: 00000000007706a0
[175372.513312] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[175372.519480] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[175372.525644] PKRU: 55555554
[175372.528550] Call Trace:
[175372.531259]  ext4_setattr+0x1fa/0x940
[175372.534679]  ? __vfs_getxattr+0x53/0x70
[175372.537936]  notify_change+0x348/0x4c0
[175372.541171]  ? chown_common+0x19f/0x1c0
[175372.544407]  chown_common+0x19f/0x1c0
[175372.547729]  do_fchownat+0x8d/0xe0
[175372.550834]  __x64_sys_fchownat+0x21/0x30
[175372.554286]  do_syscall_64+0x33/0x40
[175372.557494]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[175372.561218] RIP: 0033:0x49312a
[175372.564223] Code: e8 5b db fd ff 48 8b 7c 24 10 48 8b 74 24 18 48 8b 54 24 20 4c 8b 54 24 28 4c 8b 44 24 30 4c 8b 4c 24 38 48 8b 44 24 08 0f 05 <48> 3d 01 f0 ff ff 76 20 48 c7 44 24 40 ff ff ff ff 48 c7 44 24 48
[175372.577126] RSP: 002b:000000c01c556988 EFLAGS: 00000212 ORIG_RAX: 0000000000000104
[175372.583712] RAX: ffffffffffffffda RBX: 000000c000067800 RCX: 000000000049312a
[175372.590607] RDX: ffffffffffffffff RSI: 000000c019d45860 RDI: ffffffffffffff9c
[175372.597268] RBP: 000000c01c556a00 R08: 0000000000000100 R09: 0000000000000000
[175372.604775] R10: 0000000000002711 R11: 0000000000000212 R12: 0000000000000000
[175372.611721] R13: 0000000000000001 R14: 000000000000001b R15: ffffffffffffffff
[175372.618933] ---[ end trace 9e4c192c03053a9a ]---
[175372.623428] EXT4-fs error (device nvme7n1): __ext4_get_inode_loc_noinmem:4455: inode #2439583: block 9442233: comm kubelet: unable to read itable block
[175372.994086] Buffer I/O error on dev nvme7n1, logical block 524335, lost async page write
[175373.012310] Aborting journal on device nvme7n1-8.
[175373.015949] Buffer I/O error on dev nvme7n1, logical block 3702784, lost sync page write
[175373.022442] JBD2: Error -5 detected when updating journal superblock for nvme7n1-8.
[175373.191437] Buffer I/O error on dev nvme7n1, logical block 524336, lost async page write
[175373.291444] Buffer I/O error on dev nvme7n1, logical block 524359, lost async page write
[175373.391458] Buffer I/O error on dev nvme7n1, logical block 524366, lost async page write
[175373.491437] Buffer I/O error on dev nvme7n1, logical block 524380, lost async page write
[175373.591434] Buffer I/O error on dev nvme7n1, logical block 524382, lost async page write
[175373.791436] Buffer I/O error on dev nvme7n1, logical block 524398, lost async page write
[175373.891455] Buffer I/O error on dev nvme7n1, logical block 524404, lost async page write
[175373.991465] Buffer I/O error on dev nvme7n1, logical block 524409, lost async page write
[177392.931307] pci 0000:00:19.0: [1d0f:8061] type 00 class 0x010802
[177392.935685] pci 0000:00:19.0: reg 0x10: [mem 0x00000000-0x00003fff]
[177392.940931] pci 0000:00:19.0: BAR 0: assigned [mem 0xc0018000-0xc001bfff]
[177392.945570] nvme nvme7: pci function 0000:00:19.0
[177392.949261] nvme 0000:00:19.0: enabling device (0000 -> 0002)
[177392.959590] nvme nvme7: 2/0/0 default/read/poll queues
[177403.186975] EXT4-fs (nvme7n1): mounted filesystem with ordered data mode. Opts: (null)
[179835.282032] pci 0000:00:17.0: [1d0f:8061] type 00 class 0x010802
[179835.286496] pci 0000:00:17.0: reg 0x10: [mem 0x00000000-0x00003fff]
```

5) The `wait_on_bit` issue. I have seen this only once today but quite often in the past. Appears to occur on all platforms. This is ugly as it has an impact on productive systems (etcd lockup). It might not have been detected.

```
[167474.482775] RSP: 002b:00007f16b27f2870 EFLAGS: 00000293 ORIG_RAX: 000000000000004a
[167474.489682] RAX: ffffffffffffffda RBX: 000000000000001f RCX: 00007f16e5782a97
[167474.496476] RDX: 0000000000000000 RSI: 0000000000000000 RDI: 000000000000001f
[167474.503271] RBP: 00007f16b27f28b0 R08: 0000000000000000 R09: 000000000000000a
[167474.510158] R10: 00007f16e3f38520 R11: 0000000000000293 R12: 00007fff0ae58e9e
[167474.516991] R13: 00007fff0ae58e9f R14: 00007f16b27fe700 R15: 0000000000000000
[167474.523926] INFO: task promtail:1699003 blocked for more than 121 seconds.
[167474.529141]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[167474.536335] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[167474.543632] task:promtail        state:D stack:    0 pid:1699003 ppid:     1 flags:0x00000080
[167474.551140] Call Trace:
[167474.554315]  __schedule+0x282/0x870
[167474.557927]  ? __find_get_block+0x225/0x2c0
[167474.561884]  ? out_of_line_wait_on_bit_lock+0xb0/0xb0
[167474.566109]  schedule+0x46/0xb0
[167474.569547]  io_schedule+0x42/0x70
[167474.573018]  bit_wait_io+0xd/0x50
[167474.576410]  __wait_on_bit+0x2a/0x90
[167474.580020]  out_of_line_wait_on_bit+0x92/0xb0
[167474.584036]  ? var_wake_function+0x20/0x20
[167474.587827]  __ext4_find_entry+0x2d0/0x430
[167474.591600]  ext4_lookup+0x86/0x1f0
[167474.595226]  path_openat+0x54d/0x1070
[167474.598880]  do_filp_open+0x88/0x130
[167474.602458]  ? getname_flags.part.0+0x29/0x1a0
[167474.606426]  ? __check_object_size+0x136/0x150
[167474.610363]  do_sys_openat2+0x97/0x150
[167474.614057]  __x64_sys_openat+0x54/0x90
[167474.617816]  do_syscall_64+0x33/0x40
[167474.621462]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[167474.625774] RIP: 0033:0x4ce84a
[167474.629119] RSP: 002b:000000c000755c08 EFLAGS: 00000216 ORIG_RAX: 0000000000000101
[167474.636331] RAX: ffffffffffffffda RBX: 000000c000061000 RCX: 00000000004ce84a
[167474.643274] RDX: 0000000000080241 RSI: 000000c000b97e80 RDI: ffffffffffffff9c
[167474.650270] RBP: 000000c000755c80 R08: 0000000000000000 R09: 0000000000000000
[167474.657215] R10: 0000000000000180 R11: 0000000000000216 R12: ffffffffffffffff
[167474.664177] R13: 00000000000000f5 R14: 00000000000000f4 R15: 0000000000000100
[167474.671357] INFO: task kworker/u16:4:2782503 blocked for more than 121 seconds.
[167474.678377]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[167474.685737] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[167474.692810] task:kworker/u16:4   state:D stack:    0 pid:2782503 ppid:     2 flags:0x00004080
[167474.700220] Workqueue: writeback wb_workfn (flush-259:0)
[167474.704528] Call Trace:
[167474.707633]  __schedule+0x282/0x870
[167474.711361]  ? blk_mq_flush_plug_list+0x100/0x190
[167474.715645]  ? wbt_rqw_done+0xe0/0xe0
[167474.719192]  schedule+0x46/0xb0
[167474.722483]  io_schedule+0x42/0x70
[167474.725908]  rq_qos_wait+0xbd/0x140
[167474.729384]  ? efi_partition+0x810/0x810
[167474.733081]  ? wbt_cleanup_cb+0x20/0x20
[167474.736686]  wbt_wait+0x99/0x100
[167474.740048]  __rq_qos_throttle+0x23/0x30
[167474.743701]  blk_mq_submit_bio+0x124/0x530
[167474.747450]  submit_bio_noacct+0x3a9/0x420
[167474.751200]  ext4_io_submit+0x49/0x60
[167474.754702]  ext4_writepages+0x22e/0xfc0
[167474.758337]  ? pagevec_lookup_range_tag+0x24/0x30
[167474.762447]  ? write_cache_pages+0xaf/0x3d0
[167474.766350]  ? __wb_calc_thresh+0x120/0x120
[167474.770381]  ? generic_writepages+0x57/0x80
[167474.774232]  ? fprop_fraction_percpu+0x2b/0x70
[167474.778109]  do_writepages+0x34/0xc0
[167474.781613]  ? wb_calc_thresh+0x4f/0x70
[167474.785379]  __writeback_single_inode+0x39/0x2a0
[167474.789616]  writeback_sb_inodes+0x200/0x470
[167474.793437]  __writeback_inodes_wb+0x4c/0xe0
[167474.797348]  wb_writeback+0x1d8/0x290
[167474.800888]  wb_workfn+0x2d5/0x4d0
[167474.804291]  ? __switch_to_asm+0x42/0x70
[167474.808056]  process_one_work+0x1b6/0x350
[167474.811750]  worker_thread+0x53/0x3e0
[167474.815288]  ? process_one_work+0x350/0x350
[167474.819344]  kthread+0x11b/0x140
[167474.823125]  ? __kthread_bind_mask+0x60/0x60
[167474.827390]  ret_from_fork+0x22/0x30
[167474.830972] INFO: task etcd:2790380 blocked for more than 121 seconds.
[167474.835959]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[167474.843113] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[167474.850118] task:etcd            state:D stack:    0 pid:2790380 ppid:2782868 flags:0x00000080
[167474.857559] Call Trace:
[167474.860557]  __schedule+0x282/0x870
[167474.864020]  schedule+0x46/0xb0
[167474.867289]  io_schedule+0x42/0x70
[167474.870687]  wait_on_page_bit_common+0x116/0x3b0
[167474.874642]  ? trace_event_raw_event_file_check_and_advance_wb_err+0xf0/0xf0
[167474.879687]  wait_on_page_writeback+0x25/0x70
[167474.883590]  __filemap_fdatawait_range+0x81/0xf0
[167474.887561]  ? ext4_buffered_write_iter+0xce/0x160
[167474.891687]  ? __filemap_fdatawrite_range+0xd0/0x100
[167474.895856]  file_write_and_wait_range+0x8c/0xb0
[167474.899839]  ext4_sync_file+0x73/0x350
[167474.903418]  __x64_sys_fdatasync+0x47/0x80
[167474.907178]  do_syscall_64+0x33/0x40
[167474.910651]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[167474.914790] RIP: 0033:0x4a97d0
[167474.918084] RSP: 002b:000000c0002b0da8 EFLAGS: 00000216 ORIG_RAX: 000000000000004b
[167474.925100] RAX: ffffffffffffffda RBX: 000000c00004c500 RCX: 00000000004a97d0
[167474.931820] RDX: 0000000000000000 RSI: 0000000000000000 RDI: 000000000000000a
[167474.938586] RBP: 000000c0002b0de8 R08: 0000000000000000 R09: 0000000000000000
[167474.945314] R10: 0000000000000000 R11: 0000000000000216 R12: ffffffffffffffff
[167474.952087] R13: 00000000000000b3 R14: 00000000000000b2 R15: 0000000000000100
[167474.958916] INFO: task etcd:2790513 blocked for more than 121 seconds.
[167474.963917]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[167474.971296] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[167474.978599] task:etcd            state:D stack:    0 pid:2790513 ppid:2782868 flags:0x00000080
[167474.985974] Call Trace:
[167474.988943]  __schedule+0x282/0x870
[167474.992397]  schedule+0x46/0xb0
[167474.995665]  io_schedule+0x42/0x70
[167474.999039]  wait_on_page_bit_common+0x116/0x3b0
[167475.002970]  ? trace_event_raw_event_file_check_and_advance_wb_err+0xf0/0xf0
[167475.008158]  wait_on_page_writeback+0x25/0x70
[167475.011989]  __filemap_fdatawait_range+0x81/0xf0
[167475.015914]  ? __filemap_fdatawrite_range+0xd0/0x100
[167475.020061]  file_write_and_wait_range+0x8c/0xb0
[167475.024003]  ext4_sync_file+0x73/0x350
[167475.027602]  __x64_sys_fdatasync+0x47/0x80
[167475.031340]  do_syscall_64+0x33/0x40
[167475.034982]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[167475.039198] RIP: 0033:0x4a97d0
[167475.042609] RSP: 002b:000000c006c679b8 EFLAGS: 00000212 ORIG_RAX: 000000000000004b
[167475.049807] RAX: ffffffffffffffda RBX: 000000c00004a000 RCX: 00000000004a97d0
[167475.056589] RDX: 0000000000000000 RSI: 0000000000000000 RDI: 0000000000000006
[167475.063267] RBP: 000000c006c679f8 R08: 0000000000000000 R09: 0000000000000000
[167475.069989] R10: 0000000000000000 R11: 0000000000000212 R12: 0000000000000004
[167475.076700] R13: 0000000000000011 R14: 000000000124f29c R15: 0000000000000000
[173544.561468] pci 0000:00:19.0: [1d0f:8061] type 00 class 0x010802
[173544.566390] pci 0000:00:19.0: reg 0x10: [mem 0x00000000-0x00003fff]
[173544.572436] pci 0000:00:19.0: BAR 0: assigned [mem 0xc0018000-0xc001bfff]
[173544.577659] nvme nvme7: pci function 0000:00:19.0
[173544.581703] nvme 0000:00:19.0: enabling device (0000 -> 0002)
[173544.593607] nvme nvme7: 2/0/0 default/read/poll queues
```

6) Two issues that appear to show up on VMware only:

```
[1984684.457335] sd 0:0:0:0: [sda] tag#99 task abort on host 0, 0000000002af6841
[1984687.769162] INFO: task jbd2/sda3-8:373 blocked for more than 120 seconds.
[1984687.770538]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[1984687.772059] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[1984687.773559] task:jbd2/sda3-8     state:D stack:    0 pid:  373 ppid:     2 flags:0x00004000
[1984687.775158] Call Trace:
[1984687.775679]  __schedule+0x282/0x870
[1984687.776374]  schedule+0x46/0xb0
[1984687.777013]  jbd2_journal_commit_transaction+0x320/0x1ad0
[1984687.777165] sd 0:0:0:0: [sda] tag#114 task abort on host 0, 0000000096d2b2ce
[1984687.778078]  ? sched_clock+0x5/0x10
[1984687.778777]  ? sched_clock_cpu+0xc/0xa0
[1984687.779536]  ? psi_group_change+0x41/0x220
[1984687.780340]  ? add_wait_queue_exclusive+0x70/0x70
[1984687.781267]  ? lock_timer_base+0x61/0x80
[1984687.782043]  kjournald2+0xab/0x270
[1984687.782723]  ? add_wait_queue_exclusive+0x70/0x70
[1984687.783632]  ? load_superblock.part.0+0xb0/0xb0
[1984687.784537]  kthread+0x11b/0x140
[1984687.785260]  ? __kthread_bind_mask+0x60/0x60
[1984687.786096]  ret_from_fork+0x22/0x30
[1984687.786808] INFO: task systemd-journal:470 blocked for more than 120 seconds.
[1984687.788166]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[1984687.789692] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[1984687.791178] task:systemd-journal state:D stack:    0 pid:  470 ppid:     1 flags:0x00000320
[1984687.792752] Call Trace:
[1984687.793293]  __schedule+0x282/0x870
[1984687.793996]  schedule+0x46/0xb0
[1984687.794650]  wait_transaction_locked+0x8a/0xd0
[1984687.795520]  ? add_wait_queue_exclusive+0x70/0x70
[1984687.796433]  add_transaction_credits+0xd6/0x2a0
[1984687.797345]  ? __wake_up_common_lock+0x8a/0xc0
[1984687.798218]  start_this_handle+0xfb/0x520
[1984687.799012]  ? jbd2__journal_start+0x8d/0x1e0
[1984687.799871]  ? kmem_cache_alloc+0xed/0x1f0
[1984687.800673]  jbd2__journal_start+0xf7/0x1e0
[1984687.801526]  __ext4_journal_start_sb+0xf3/0x110
[1984687.802415]  ext4_dirty_inode+0x3d/0x80
[1984687.803178]  __mark_inode_dirty+0x192/0x2f0
[1984687.804005]  generic_update_time+0x68/0xc0
[1984687.804834]  file_update_time+0x123/0x140
[1984687.805646]  ext4_page_mkwrite+0x87/0x630
[1984687.806438]  do_page_mkwrite+0x55/0xb0
[1984687.807184]  do_wp_page+0x2a1/0x470
[1984687.807881]  handle_mm_fault+0x1122/0x1bf0
[1984687.808684]  do_user_addr_fault+0x1b8/0x3f0
[1984687.809534]  exc_page_fault+0x78/0x160
[1984687.810283]  ? asm_exc_page_fault+0x8/0x30
[1984687.811092]  asm_exc_page_fault+0x1e/0x30
[1984687.811897] RIP: 0033:0x7f3b9a297749
[1984687.812632] RSP: 002b:00007ffe3c825bf0 EFLAGS: 00010246
[1984687.813656] RAX: 00007f3b95b6b4f8 RBX: 0000560f61cc5080 RCX: 00007ffe3c825bb0
[1984687.815198] RDX: 0000000000298d2c RSI: 0000000000000000 RDI: 00007f3b976c3910
[1984687.816722] RBP: 0000000000000000 R08: 0000000000000000 R09: 00007ffe3c823a60
[1984687.818090] R10: 0000000000000000 R11: 0000000000000293 R12: 0000000000080802
[1984687.819455] R13: 00000000ffffffff R14: 0000560f61be7fa0 R15: 00000000fffffffa
[1984687.820834] INFO: task containerd:3268 blocked for more than 120 seconds.
[1984687.822132]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[1984687.823637] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[1984687.825129] task:containerd      state:R  running task     stack:    0 pid: 3268 ppid:  3213 flags:0x00000080
[1984687.826988] Call Trace:
[1984687.827504]  __schedule+0x282/0x870
[1984687.828199]  schedule+0x46/0xb0
[1984687.828847]  rwsem_down_write_slowpath+0x242/0x4c0
[1984687.829795]  ? strncpy_from_user+0x4e/0x140
[1984687.830620]  lock_rename+0xac/0xb0
[1984687.831299]  do_renameat2+0x268/0x540
[1984687.832024]  __x64_sys_renameat+0x1d/0x20
[1984687.832814]  do_syscall_64+0x33/0x40
[1984687.833549]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[1984687.834557] RIP: 0033:0xaea38a
[1984687.835181] RSP: 002b:000000c001364fb8 EFLAGS: 00000212 ORIG_RAX: 0000000000000108
[1984687.836614] RAX: ffffffffffffffda RBX: 000000c00004e800 RCX: 0000000000aea38a
[1984687.838125] RDX: ffffffffffffff9c RSI: 000000c0027f45a0 RDI: ffffffffffffff9c
[1984687.839726] RBP: 000000c001365058 R08: 0000000000000000 R09: 0000000000000000
[1984687.841216] R10: 000000c0027f4630 R11: 0000000000000212 R12: 000000c0027f4630
[1984687.842577] R13: 0000000000000001 R14: 000000c0014f96c0 R15: ffffffffffffffff
[1987609.226442] vmw_pvscsi: msg type: 0x0 - MSG RING: 51/50 (5)
[1987609.227732] vmw_pvscsi: msg: device added at scsi0:3:0
[1987609.229057] scsi 3:0:3:0: Direct-Access     VMware   Virtual disk     2.0  PQ: 0 ANSI: 6
[1987609.237257] sd 3:0:3:0: Attached scsi generic sg18 type 0
[1987609.237380] sd 3:0:3:0: [sds] 41943040 512-byte logical blocks: (21.5 GB/20.0 GiB)
[1987609.240393] sd 3:0:3:0: [sds] Write Protect is off
[1987609.241352] sd 3:0:3:0: [sds] Mode Sense: 31 00 00 00
[1987609.241420] sd 3:0:3:0: [sds] Cache data unavailable
[1987609.242457] sd 3:0:3:0: [sds] Assuming drive cache: write through
[1987609.286610] sd 3:0:3:0: [sds] Attached SCSI disk
```

and

```
[2389065.704434] INFO: task flb-pipeline:243478 blocked for more than 120 seconds.
[2389065.705895]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[2389065.707415] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[2389065.708906] task:flb-pipeline    state:D stack:    0 pid:243478 ppid:243411 flags:0x00000080
[2389065.710497] Call Trace:
[2389065.711012]  __schedule+0x282/0x870
[2389065.711700]  schedule+0x46/0xb0
[2389065.712343]  io_schedule+0x42/0x70
[2389065.713027]  wait_on_page_bit_common+0x116/0x3b0
[2389065.713913]  ? trace_event_raw_event_file_check_and_advance_wb_err+0xf0/0xf0
[2389065.715241]  wait_on_page_writeback+0x25/0x70
[2389065.716100]  __filemap_fdatawait_range+0x81/0xf0
[2389065.716987]  ? __filemap_fdatawrite_range+0xd0/0x100
[2389065.717985]  file_write_and_wait_range+0x8c/0xb0
[2389065.718877]  ext4_sync_file+0x73/0x350
[2389065.719613]  __x64_sys_fsync+0x34/0x60
[2389065.720366]  do_syscall_64+0x33/0x40
[2389065.721073]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[2389065.722035] RIP: 0033:0x7fa448e7da97
[2389065.722736] RSP: 002b:00007fa4165f2870 EFLAGS: 00000293 ORIG_RAX: 000000000000004a
[2389065.724153] RAX: ffffffffffffffda RBX: 000000000000001d RCX: 00007fa448e7da97
[2389065.725483] RDX: 0000000000000000 RSI: 0000000000000000 RDI: 000000000000001d
[2389065.726817] RBP: 00007fa4165f28b0 R08: 0000000000000000 R09: 000000000000000a
[2389065.728187] R10: 00007fa447532aa0 R11: 0000000000000293 R12: 00007ffd3f285b4e
[2389065.729523] R13: 00007ffd3f285b4f R14: 00007fa4165fe700 R15: 0000000000000000
[2389105.649462] sd 0:0:0:0: [sda] tag#255 task abort on host 0, 000000009424cec1
```

7) This one is on Azure. Seen once. The impact is unclear:

```
[4312429.673267] ata2.01: NODEV after polling detection
[4312430.056894] ata2.01: NODEV after polling detection
[4312430.448105] ata2.01: NODEV after polling detection
[4312430.515347] ata2: EH pending after 5 tries, giving up
[4312430.995756] ata2.01: NODEV after polling detection
[4314831.651665] ata2.01: NODEV after polling detection
[4318507.303850] ata2.01: NODEV after polling detection
[4357012.088356] ata2.01: NODEV after polling detection
[4399938.604702] ata2.01: NODEV after polling detection
[4399939.277799] ata2.01: NODEV after polling detection
[4399954.438990] ata2.01: NODEV after polling detection
[4399954.818198] ata2.01: NODEV after polling detection
[4399955.197092] ata2.01: NODEV after polling detection
[4399955.585566] ata2.01: NODEV after polling detection
[4399956.006584] ata2.01: NODEV after polling detection
[4399956.071435] ata2: EH pending after 5 tries, giving up
[4399956.667956] ata2.01: NODEV after polling detection
[4399957.047611] ata2.01: NODEV after polling detection
```

8) Mysteriously dockerd is blocked here (AWS, GL576.3):

```
[4404428.513547] INFO: task dockerd:37144 blocked for more than 120 seconds.
[4404428.518481]       Not tainted 5.10.0-garden-cloud-amd64 #1 Debian 5.10.93-0gardenlinux1
[4404428.525571] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[4404428.532452] task:dockerd         state:D stack:    0 pid:37144 ppid:     1 flags:0x00004080
[4404428.539564] Call Trace:
[4404428.542498]  __schedule+0x282/0x870
[4404428.545891]  ? blk_mq_sched_insert_requests+0x9a/0xe0
[4404428.549913]  schedule+0x46/0xb0
[4404428.553145]  rwsem_down_write_slowpath+0x242/0x4c0
[4404428.557044]  ? blk_finish_plug+0x36/0x50
[4404428.560543]  sync_inodes_sb+0xba/0x2c0
[4404428.564009]  ? __filemap_fdatawrite_range+0xd0/0x100
[4404428.567926]  sync_filesystem+0x5a/0x90
[4404428.571371]  ovl_sync_fs+0x55/0x90 [overlay]
[4404428.575048]  sync_filesystem+0x74/0x90
[4404428.578522]  generic_shutdown_super+0x22/0x100
[4404428.582318]  kill_anon_super+0x14/0x30
[4404428.585776]  deactivate_locked_super+0x31/0x90
[4404428.589585]  cleanup_mnt+0x12d/0x190
[4404428.592964]  task_work_run+0x65/0xa0
[4404428.596359]  exit_to_user_mode_prepare+0x111/0x120
[4404428.600292]  syscall_exit_to_user_mode+0x28/0x140
[4404428.604149]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[4404428.608125] RIP: 0033:0x55a2e421a45b
[4404428.611513] RSP: 002b:000000c002c8c9a0 EFLAGS: 00000202 ORIG_RAX: 00000000000000a6
[4404428.618228] RAX: 0000000000000000 RBX: 000000c000065800 RCX: 000055a2e421a45b
[4404428.624770] RDX: 0000000000000000 RSI: 0000000000000002 RDI: 000000c00209c850
[4404428.631370] RBP: 000000c002c8ca00 R08: 00007f6974a90f01 R09: 0000000000000000
[4404428.637912] R10: 00007f694c02f1b0 R11: 0000000000000202 R12: 000000c00209c850
[4404428.644484] R13: 0000000000000001 R14: 000000c000f28d00 R15: ffffffffffffffff
```

**What you expected to happen**:

No such messages.

**How to reproduce it (as minimally and precisely as possible)**:

Run a large Gardener infrastructure and wait.

**Anything else we need to know**:

**Environment**:

**Comment by Vincinator** [comment]
Closing this issue. Created a new issue #1603 to do develop tools and define a setup, to help us get hands on kernel messages.
